mypy/types.py

```diff
@@ -1775,7 +1775,7 @@ def copy_modified(
         self: CT,
         arg_types: Bogus[Sequence[Type]] = _dummy,
         arg_kinds: Bogus[list[ArgKind]] = _dummy,
-        arg_names: Bogus[list[str | None]] = _dummy,
+        arg_names: Bogus[Sequence[str | None]] = _dummy,
```

Comment:
What is the reason for this change? I'd prefer to avoid touching the mypy core in this PR.

Reply:
Because of list's invariance.  In `CallableType.__init__`, we copy `self.arg_names = list(arg_names)` so we can safely make it a `Sequence[str | None]`.

Otherwise, this fails since it passes `list[str]`:
```python
arg_names=["inst"] + attrs_init_type.arg_names[1:],
```

Reply:
Broke out into #14840 — can merge then update this branch.